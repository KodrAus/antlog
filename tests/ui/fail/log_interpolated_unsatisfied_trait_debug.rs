#![feature(stmt_expr_attributes, proc_macro_hygiene)]

#[macro_use]
extern crate emit;

// Does not implement `Display`
struct Input;

fn main() {
    info!("Text \"and\" {#[with_debug] a: Input} and more");
}
