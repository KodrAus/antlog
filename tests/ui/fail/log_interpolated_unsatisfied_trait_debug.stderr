error[E0277]: `Input` doesn't implement `Debug`
  --> $DIR/log_interpolated_unsatisfied_trait_debug.rs:10:26
   |
10 |     emit!("Text \"and\" {#[debug] a: Input} and more");
   |                          ^^^^^^^^^^^^^^^^^ `Input` cannot be formatted using `{:?}`
   |
   = help: the trait `Debug` is not implemented for `Input`
   = note: add `#[derive(Debug)]` or manually implement `Debug`
   = note: required because of the requirements on the impl of `emit_rt::capture::Capture<emit_rt::capture::CaptureDebug>` for `Input`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
